### ISSUE TITLE
use it in combination with @RestController to return 404

### ISSUE THREAD

**Issue by coderzc** [dabla/grizzly-spring-boot-starter#5]
I use it in combination with @RestController to return 404. Is it because of my usage problem?
```java
@RestController
public class HomeController {

    @GetMapping("/index")
    public String index() {
        return "helloWorld";
    }

}
```
```
server.port=8081
server.servlet.context-path=/
```
![image](https://user-images.githubusercontent.com/26179648/112108907-d1e5b600-8beb-11eb-89aa-63ea535adfce.png)

**Comment by dabla** [comment]
Hello, I've never tested it with Spring Rest controllers but tried an integration test to reproduce your case and I can confirm it doesn't work.  The rest controllers are well registered but it seems like grizzly doesn't match the request with the rest controller.

I will try to investigate further or hopefully someone from the Spring community can help me on this one.

An alternative solution is to use the Jersey implementation which is in fact the reference implementation of JAX-RS in Java.

You can find an example on how to do this in the test package [here](https://github.com/dabla/grizzly-spring-boot-starter/tree/master/grizzly-spring-boot-starter/src/test/java/be/dabla/boot/grizzly/hello).

**Comment by dabla** [comment]
The issue using Spring Rest controllers in combination with Jersey Rest controllers will be fixed in version 2.6.  To make sure both don't conflict with each other it is advised to define as specific path for the spring.jersey.application-path property.

For example:

`spring.jersey.application-path=/api`
